### PR TITLE
Send LSP error when GHC cannot be found

### PR DESCRIPTION
**Pull request by hasufell**

### CODE REVIEW

File: bindist/wrapper.in [hunk 1]
```diff
@@ -23,10 +23,13 @@ instruction_msg() {
...
 }
 
-err_fix() {
-	instruction_msg ""
-	instruction_msg "Consider installing ghc-${GHC_VERSION} via ghcup"
-	instruction_msg "or build HLS from source."
+err_exit() {
+	msg="Couldn't find a working/matching GHC installation. Consider installing ghc-${GHC_VERSION} via ghcup or build HLS from source."
+	json="{\"jsonrpc\":\"2.0\", \"method\":\"window/showMessage\", \"params\": {\"type\": 1, \"message\": \"${msg}\"}}"
+	printf "%s\r\n" "Content-Length: $(printf "%s" "${json}" | wc -c)"
```
Comment (hasufell) [inline bindist/wrapper.in:29]: I hope `wc` exists on all platforms? Maybe we should hardcode ^^

**Review by pepeiborra — APPROVED** [review]
It will spam 5 times like @fendor says, but it's hard to improve until we merge the bindist wrapper into the hls wrapper and make the latter able to send a `WindowShowMessageRequest` ... which might take a while.

**Review by fendor — APPROVED** [review]

**Comment by michaelpj** [comment]
Is there a reason to do this for this error but not for the others?

**Comment by hasufell** [comment]
> Is there a reason to do this for this error but not for the others?

We want to do it for the final error only. The script tries multiple methods to find a GHC version. Reporting all intermediate errors is gonna be excessive.

**Comment by michaelpj** [comment]
I see, so the other `err_*` functions are not in fact reporting errors.

Anyway, if you're into the idea, there's also `window/logMessage`.

**Pull request merged**